### DISCUSSION TITLE
Storing constants in the API filter

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2346] [Q&A]
I'm thinking that it would be very useful to have access to constants in the API filter rules.  This could be loaded in different ways.    For myself I'd load it as a framework.  This would allow me to get Golang constant values into the UI.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the use case for this. Could you give an example how you would use it?

**Comment by dschissler** [comment]
I have a filter rule like this.

`@request.auth.group.rank >= 50`

It looks like `name: admin, rank: 50,` .  Normal looks like: `name: normal, rank: 10`.

It's a nice way of having some groups be higher than others, while still being able to refer to it with something like `normal`.

Or even just the `= 'admin` could be a constant.

I'm saying to be able to load it different ways because it could be useful for pre-compiled binary and as a framework.  For as a framework you might want to sync it with Golang constants values by loading them on boot.

**Reply by ganigeorgiev** [reply]
I don't understand the rank example. If you want conditional rank behavior based on some other field, I think it would be better to maintain it inside the rules.

I could understand the constant use case, but at the same time I don't see much value in this since most likely you'll already have the value as part of a `select` option and it might be better to invest time in enum-like behavior based on the field options rather than some obscure constants loading.

**Reply by dschissler** [reply]
> it might be better to invest time in enum-like behavior based on the field options

It's an extra join, but perhaps I'll try that.
